compiler/rustc_codegen_cranelift/src/constant.rs

```diff
@@ -138,8 +140,11 @@ pub(crate) fn codegen_const_value<'tcx>(
                 let base_addr = match fx.tcx.global_alloc(alloc_id) {
                     GlobalAlloc::Memory(alloc) => {
                         if alloc.inner().len() == 0 {
-                            assert_eq!(offset, Size::ZERO);
-                            fx.bcx.ins().iconst(fx.pointer_type, alloc.inner().align.bytes() as i64)
+                            let val = alloc.inner().align.bytes().wrapping_add(offset.bytes());
+                            fx.bcx.ins().iconst(
+                                fx.pointer_type,
+                                fx.tcx.truncate_to_target_usize(val) as i64,
+                            )
```

Comment:
The proper fix would have been to remove the `assert_eq!()` here and to do the `truncate_to_target_usize` at the `fx.bcx.ins().iadd_imm(base_addr, i64::try_from(offset.bytes()).unwrap())` below. As is this still ICEs. Will push a fix to the cg_clif repo.